From f8c841e3fd2f8d7f1c5de4ad1a003e9725d37499 Mon Sep 17 00:00:00 2001
From: Ankit Agrawal <ankit.agarwal@intel.com>
Date: Thu, 18 May 2023 16:19:38 +0530
Subject: [PATCH] WA-Fixed Build Error Releasetools.

Build error-:
File "build/tools/releasetools/common.py", line 4104
    def ParseUpdateEngineConfig(path: str):
                                    ^
SyntaxError: invalid syntax

File "build/tools/releasetools/common.py", line 4123
    f"{path} is an invalid update_engine config, missing PAYLOAD_MAJOR_VERSION {data}")
                                                                                     ^
SyntaxError: invalid syntax

File "build/tools/releasetools/common.py", line 4124
    f"{path} is an invalid update_engine config, missing PAYLOAD_MINOR_VERSION {data}")
                                                                                     ^
SyntaxError: invalid syntax

Seems due to python version. Above seems not supported in python2.

Tracked-On: OAM-106853
Signed-off-by: Ankit Agrawal <ankit.agarwal@intel.com>
---
 tools/releasetools/common.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/tools/releasetools/common.py b/tools/releasetools/common.py
index 68eb36f678..6db1d14782 100644
--- a/tools/releasetools/common.py
+++ b/tools/releasetools/common.py
@@ -4101,7 +4101,7 @@ def IsSparseImage(filepath):
     return fp.read(4) == b'\x3A\xFF\x26\xED'
 
 
-def ParseUpdateEngineConfig(path: str):
+def ParseUpdateEngineConfig(path):
   """Parse the update_engine config stored in file `path`
   Args
     path: Path to update_engine_config.txt file in target_files
@@ -4117,9 +4117,9 @@ def ParseUpdateEngineConfig(path: str):
     major = re.search(r"PAYLOAD_MAJOR_VERSION=(\d+)", data)
     if not major:
       raise ValueError(
-          f"{path} is an invalid update_engine config, missing PAYLOAD_MAJOR_VERSION {data}")
+          "{path} is an invalid update_engine config, missing PAYLOAD_MAJOR_VERSION {data}" + path)
     minor = re.search(r"PAYLOAD_MINOR_VERSION=(\d+)", data)
     if not minor:
       raise ValueError(
-          f"{path} is an invalid update_engine config, missing PAYLOAD_MINOR_VERSION {data}")
+          "{path} is an invalid update_engine config, missing PAYLOAD_MINOR_VERSION {data}" + path)
     return (int(major.group(1)), int(minor.group(1)))
-- 
2.17.1

